src/test/scala/chiselTests/experimental/hierarchy/InstanceSpec.scala

```diff
@@ -1066,4 +1066,32 @@ class InstanceSpec extends ChiselFunSpec with Utils {
       getFirrtlAndAnnos(new HasMultipleTypeParamsInside, Seq(aspect))
     }
   }
+  it("11.1 suggestName for Instances") {
+    class Top extends Module {
+      val definition = Definition(new AddOne)
+      val inst0 = Instance(definition)
+      inst0.suggestName("potato")
+    }
+    val (chirrtl, _) = getFirrtlAndAnnos(new Top)
+    chirrtl.serialize should include("inst potato of AddOne")
+  }
+  it("11.2 suggestName at instantiation") {
+    class Top extends Module {
+      val k = Instance(Definition(new AddOne)).suggestName("potato")
+    }
+    val (chirrtl, _) = getFirrtlAndAnnos(new Top)
+    chirrtl.serialize should include("inst potato of AddOne")
+  }
+  it("11.2 suggestName with sanitization") {
+    class Top extends Module {
+      val definition = Definition(new AddOne)
+      val inst0 = Instance(definition)
+      val inst1 = Instance(definition)
+      inst0.suggestName("potato")
+      inst1.suggestName("potato")
```

Comment:
can you show a test with two different things that are not the same `Definition`, `Instance` which also name conflict and successfully sanitize?
```
val inst0 = Module(new AddOne()).suggestName("potato")
val inst1 = Instance(definition).suggestName("potato")
```